File: Objc/Direct_Pay_CCVForm/Direct_Pay_CCV_Form/Direct_Pay_CCV_Form/ViewController.m
//
//  ViewController.m
//  Direct_Pay_CCV_Form
//
//  Created by Cherri Kevin on 5/13/21.
//

#import "ViewController.h"
#import <TPDirect/TPDirect.h>

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIView *ccvFormView;


@property (strong, nonatomic) TPDCcvForm * tpdCcvForm;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.tpdCcvForm = [TPDCcvForm setupCCVWithContainer:self.ccvFormView];
    
    [self.tpdCcvForm onCCVFormUpdated:^(TPDStatus * _Nullable status) {
        NSLog(@"isCanGetCCVPrime : %d", [status isCanGetCCVPrime]);
    }];
    
    
}
- (IBAction)getCCVPrime:(id)sender {
    TPDCcv *tpdCcv = [TPDCcv setup:self.tpdCcvForm];
    
    [[[tpdCcv onSuccessCallback:^(NSString * _Nonnull prime) {
        NSLog(@"CCV Prime : %@", prime);
    }] onFailureCallback:^(NSInteger status, NSString * _Nonnull message) {
        NSLog(@"status : %ld, message : %@", status, message);
    }] getPrime];
}



@end

File: Swift/Direct_Pay_CCVForm/Direct_Pay_CCVForm/Direct_Pay_CCVForm/ViewController.swift
//
//  ViewController.swift
//  Direct_Pay_CCVForm
//
//  Created by Cherri Kevin on 5/13/21.
//

import UIKit
import TPDirect

class ViewController: UIViewController {

    @IBOutlet weak var ccvFormView: UIView!
    var tpdCcvForm : TPDCcvForm!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.tpdCcvForm = TPDCcvForm.setupCCV(withContainer: self.ccvFormView)
        
        self.tpdCcvForm.onCCVFormUpdated { (status) in
            print("isCanGetCCVPrime : \(status!.isCanGetCCVPrime())")
        }
        
    }

    
    @IBAction func getCCVPrime(_ sender: Any) {
        let tpdCcv = TPDCcv.setup(self.tpdCcvForm)
        
        tpdCcv.onSuccessCallback { (prime) in
            print("prime : \(prime)")
        }.onFailureCallback { (status, message) in
            print("status : \(status), message : \(message)")
        }.getPrime()
    }
    
}

